builder/builder.go

```diff
@@ -81,6 +81,10 @@ type Builder struct {
 	NumFullfileWorkers int
 	NumDeltaWorkers    int
 	NumChrootWorkers   int
+
+	// Parsed versions.
+	MixVersion      uint32
+	UpstreamVersion uint32
```

Comment:
I like what you're doing in this commit, I just don't like the naming convention.  `MixVer` == string while `MixVersion` == uint32 will definitely cause confusion later on.  If we want to preserve both (which is fine), perhaps `MixVer` and `MixVerInt` or `MixVerUint32`?

Reply:
Chose MixVerUint32 and UpstreamVerUint32. I prefer not touch the existing variable for this change.

Please consider revamping those names later: MixVersion and MixVersionStr seems good names for the uint and string variables.